src/game/GameProcLogIn.cpp

```diff
@@ -123,6 +123,12 @@ void CGameProcLogIn::Init() {
     lstrcat(szIniPath, "Server.Ini");
     int iServerCount = GetPrivateProfileInt("Server", "Count", 0, szIniPath);
 
+    char szRegistrationSite[_MAX_PATH];
+    memset(szRegistrationSite, 0, sizeof(szRegistrationSite));
+
+    GetPrivateProfileString("Join", "Registration site", "", szRegistrationSite, _MAX_PATH, szIniPath);
+    m_szRegistrationSite = std::string(szRegistrationSite);
+
```

Comment:
Note that allocating resources for something that happens only once during runtime may be wasteful. Generally in the Login UI screen, we have a Join button that when we click it, it opens the loaded URL from `Server.ini` and closes the game. This said, I suggest implementing it the way it's implemented officially in the v1264 client in `CUILogIn::ReceiveMessage`:
```c
      else if ( a_pSender == (CN3UIBase *)this->m_pBtn_Join )
      {
        LOBYTE(v17.vft) = v16;
        std::string::_Tidy(&v17, 0);
        v6 = strlen(&Default);
        if ( std::string::403250(&v17, v6, 1) )
        {
          qmemcpy(v17.buffer, &Default, v6);
          std::string::_Eos(v6);
        }
        String1[0] = Default;
        v25 = 2;
        memset(&String1[1], 0, 256u);
        v22 = 0;
        v23 = 0;
        v7 = lpString2;
        if ( !lpString2 )
          v7 = NewFilename;
        lstrcpyA(String1, v7);
        lstrcatA(String1, aServerIni_0);
        memset(ReturnedString, 0, sizeof(ReturnedString));
        GetPrivateProfileStringA(aJoin, aRegistrationSi, &Default, ReturnedString, 260u, String1);
        v8 = strlen(ReturnedString) + 1;
        v9 = v8 - 1;
        if ( std::string::403250(&v17, v8 - 1, 1) )
        {
          qmemcpy(v17.buffer, ReturnedString, v9);
          v17.length = v8 - 1;
          *((_BYTE *)v17.buffer + v9) = 0;
        }
        if ( v17.length )
        {
          PostQuitMessage(0);
          buffer = (const CHAR *)v17.buffer;
          if ( !v17.buffer )
            buffer = NewFilename;
          ShellExecuteA(0, Operation, buffer, 0, 0, 1);
        }
        if ( v17.buffer )
        {
          v11 = *((_BYTE *)v17.buffer - 1);
          if ( !v11 || v11 == -1 )
            operatorDeleteArray((char *)v17.buffer - 1);
          else
            *((_BYTE *)v17.buffer - 1) = v11 - 1;
        }
      }
      return 1;
    }
```

It looks like a lot is happening here, just because STL code produce a lot of assembly instructions, but generally it does something very similar to what you do here, just in the mentioned function.
```suggestion
```

Reply:
Moved things to CUILogin::ReceiveMessage in this commit: https://github.com/ko4life-net/ko/pull/131/commits/a0a7c015504a8ca24a1b5aad8a8ca1c7d0fa1300